### PR TITLE
feat:  get and set prompts mixin

### PR DESCRIPTION
**Pull request by shahules786**
```python

from ragas.experimental.testset.simulators import AbstractQuestionSimulator
simulator = AbstractQuestionSimulator()
prompts = simulator.get_prompts()
prompts['common_theme_prompt'].instruction += "\n Ouptut only JSON"
simulator.set_prompts(common_theme_prompt=prompts['common_theme_prompt'])```

### CODE REVIEW

File: src/ragas/experimental/testset/simulators/base.py [hunk 1]
```diff
@@ -54,3 +56,23 @@ async def generate_scenarios(
...
+    def set_prompts(self, **prompts):
+        available_prompts = self.get_prompts()
+        for key, value in prompts.items():
+            if key not in available_prompts:
+                raise ValueError(
+                    f"Prompt with name '{key}' does not exist in the simulator {self.name}. Use get_prompts() to see available prompts."
+                )
+            if not isinstance(value, Prompt):
+                raise ValueError(
+                    f"Prompt with name '{key}' must be an instance of 'Prompt'"
```
Comment (jjmachan) [inline src/ragas/experimental/testset/simulators/base.py:76]:
```suggestion
                    f"Prompt with name '{key}' must be an instance of 'ragas.prompt.PydanticPrompt'"
```

**Review by jjmachan — CHANGES_REQUESTED** [review]
This should be a mixin because we are using this in 3 places in total. 

take a look at this to see it in action: https://www.perplexity.ai/search/write-me-a-tutorial-on-python-Bxu55LqmS1ivIV8pRW8TSw

**Comment by jjmachan** [comment]
if you make it a mixin, add it to metrics too

File: src/ragas/experimental/testset/simulators/base.py (hunk 1 above)
Comment (shahules786) [reply src/ragas/experimental/testset/simulators/base.py:76]: I see, thanks I didn't know that this is the use of Mixins.

**Comment by shahules786** [comment]
> This should be a mixin because we are using this in 3 places in total.
> 
> take a look at this to see it in action: https://www.perplexity.ai/search/write-me-a-tutorial-on-python-Bxu55LqmS1ivIV8pRW8TSw

I see, thanks I didn't know that this is the use of Mixins.

File: src/ragas/experimental/testset/transforms/base.py [hunk 2]
```diff
@@ -184,6 +190,26 @@ class LLMBasedExtractor(Extractor):
...
+            if key not in available_prompts:
+                raise ValueError(
+                    f"Prompt with name '{key}' does not exist in the extractor. Use get_prompts() to see available prompts."
+                )
+            if not isinstance(value, Prompt):
+                raise ValueError(
+                    f"Prompt with name '{key}' must be an instance of 'Prompt'"
+                )
+            setattr(self, key, value)
+
```
Comment (jjmachan) [inline src/ragas/experimental/testset/transforms/base.py:212]: remove this right?

**Review by jjmachan — APPROVED** [review]
have to remove transforms/base.py

**Pull request merged**